src/django_components/component_media.py

```diff
@@ -611,20 +620,39 @@ def get_template_data(self, args, kwargs, slots, context):
     # Effectively, even if the Component class defined `js_file` (or others), at "runtime" the `js` attribute
     # will be set to the content of the file.
     # So users can access `Component.js` even if they defined `Component.js_file`.
-    comp_media.template = _get_asset(
+    template_str, template_obj = _get_asset(
         comp_cls,
         comp_media,
         inlined_attr="template",
         file_attr="template_file",
         comp_dirs=comp_dirs,
-        type="template",
-    )
-    comp_media.js = _get_asset(
-        comp_cls, comp_media, inlined_attr="js", file_attr="js_file", comp_dirs=comp_dirs, type="static"
-    )
-    comp_media.css = _get_asset(
-        comp_cls, comp_media, inlined_attr="css", file_attr="css_file", comp_dirs=comp_dirs, type="static"
     )
+    comp_media.template = template_str
+
+    js_str, _ = _get_asset(comp_cls, comp_media, inlined_attr="js", file_attr="js_file", comp_dirs=comp_dirs)
+    comp_media.js = js_str
+
+    css_str, _ = _get_asset(comp_cls, comp_media, inlined_attr="css", file_attr="css_file", comp_dirs=comp_dirs)
+    comp_media.css = css_str
+
+    # If `Component.template` or `Component.template_file` were explicitly set on this class,
+    # then Template instance was already created.
+    #
+    # Otherwise, search for Template instance in parent classes, and make a copy of it.
+    if not isinstance(template_obj, Unset):
```

Comment:
This is the new logic that fixes the bug - if the current Component class doesn't define any template, then look up the parent tree to see if anyone defined them. If yes, create a copy of Template. if no, then this child component has no template either.